src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationReader.cs

```diff
@@ -975,7 +975,7 @@ private SoapArrayInfo ParseArrayType(string value)
                 throw new ArgumentException(SR.Format(SR.XmlEmptyArrayType, CurrentTag()), nameof(value));
             }
 
-            char[] chars = value.ToCharArray();
+            ReadOnlySpan<char> chars = value.AsSpan();
```

Comment:
Why did this need ToCharArray to begin with? i.e. why can't the code below just use `value` (as a string)?

Reply:
Code changed to use sting[] instead of ToCharArray or ReadOnlySpan.